dhall-json/src/Dhall/JSON.hs

```diff
@@ -624,12 +624,12 @@ parseOmission :: Parser (Value -> Value)
 parseOmission =
         Options.Applicative.flag'
             omitNull
-            (   Options.Applicative.long "omitNull"
+            (   Options.Applicative.long "omit-null"
```

Comment:
I'm not sure whether it's worth breaking compatibility here. `--omitNull` is the default setting, and I was assuming that we could get rid of it soon.

Reply:
> `--omitNull` is the default setting

I have a question about it. [This parser](https://github.com/dhall-lang/dhall-haskell/blob/master/dhall-json/src/Dhall/JSON.hs#L649) looks like one that always return `omitNull`  because of `flag` (that always succeed) as a first Alternative.

Reply:
Do you mean this bit?

https://github.com/dhall-lang/dhall-haskell/blob/90315e9eda862d70e01fd3c960854e7cddf3cc52/dhall-json/src/Dhall/JSON.hs#L639-L642

`omitNull` is the default value here, so if we pass `--preserveNull`, the result is `id`.

http://hackage.haskell.org/package/optparse-applicative-0.15.1.0/docs/Options-Applicative-Builder.html#v:flag

Reply:
This parser returns a function in any case. It can't fail. So
```haskell
parseNullPreservation <|> smth
```
will never try to use `smth`.

Reply:
Oh, yes, you're completely right! Good catch! :)

Can you fix this so `--omitNull` and `--omitEmpty` still have an effect when passed?

Reply:
Ok!

Reply:
As I can see, for now `omit-null` is complete useless: the `omitNull` is a default value for `preserve-null` flag. So this two flags are opposites.

Reply:
If default behavior is omitting of nulls, I propose to remove `--omit-null` as redundant.

Reply:
Just removed `--omit-nulls`.